tests/Helpers.php

```diff
@@ -14,6 +14,17 @@
 // Mock escaping function.
 Functions\when('wp_kses_post')->returnArg();
 
+// Mock json success and error handlers.
+Functions\when('wp_send_json_success')->alias(function(...$args) {
+    echo json_encode($args);
+});
+Functions\when('wp_send_json_error')->alias(function(...$args) {
+    echo json_encode($args);
+});
```

Comment:
```suggestion
Functions\when('wp_send_json_error')->alias(function(...$args) {
    $response = array( 'success' => false );
 
    if ( isset( $data ) ) {
        if ( is_wp_error( $data ) ) {
            $result = array();
            foreach ( $data->errors as $code => $messages ) {
                foreach ( $messages as $message ) {
                    $result[] = array(
                        'code'    => $code,
                        'message' => $message,
                    );
                }
            }
 
            $response['data'] = $result;
        } else {
            $response['data'] = $data;
        }
    }
    
    echo json_encode($response, $options);
});
```

Same as above: https://developer.wordpress.org/reference/functions/wp_send_json_error/. What do you think? It's unlikely these functions will change much and now they behave more correctly.

Reply:
Fine by me, but, as mentioned above, I am not testing those functions per se. Using real functions would make these integration and not unit tests. That's why I kept them simple.